From 72cb087d9f0bf66e278c50e28b0e3fb64b1000e5 Mon Sep 17 00:00:00 2001
From: Ariel Aviad <aaviad@habana.ai>
Date: Thu, 06 Mar 2025 14:40:28 +0200
Subject: [PATCH] [SW-214425] hl-smi: info: rename power_limit to eeprom_power_max

Rename in 'struct hl_smi_aip_info' the field power_limit
to eeprom_power_max.
This is because it holds the EEPROM max power field, and not
the power limit.
Though the rename will add clarity to the code.

Change-Id: I7b477ce5e34c17960495f4f968cb217ef66d0b66
Signed-off-by: Ariel Aviad <aaviad@habana.ai>
---

diff --git a/include/hl-smi.h b/include/hl-smi.h
index 6ffa437..d718f40 100644
--- a/include/hl-smi.h
+++ b/include/hl-smi.h
@@ -207,7 +207,7 @@
 	unsigned int used_mem;
 	unsigned int free_mem;
 	unsigned int sys_mem;
-	char power_limit[HL_FIELD_MAX_SIZE];
+	char eeprom_power_max[HL_FIELD_MAX_SIZE];
 	char frequency_limit[HL_FIELD_MAX_SIZE];
 	char model_number[HL_FIELD_MAX_SIZE];
 	char serial_number[HL_FIELD_MAX_SIZE];
diff --git a/src/eeprom.c b/src/eeprom.c
index 8a53817..8fdae71 100644
--- a/src/eeprom.c
+++ b/src/eeprom.c
@@ -213,7 +213,7 @@
 	aip_info->product_num = 0;
 	aip_info->model_number[0] = '\0';
 	aip_info->serial_number[0] = '\0';
-	aip_info->power_limit[0] = '\0';
+	aip_info->eeprom_power_max[0] = '\0';
 	aip_info->frequency_limit[0] = '\0';
 	aip_info->pcb_assembly_ver[0] = '\0';
 	aip_info->pcb_ver[0] = '\0';
@@ -237,7 +237,7 @@
 	layout_field = find_field(aip_info->layout, "Max Power [Watt]");
 	if (layout_field) {
 		parse_field_to_int(&power, layout_field);
-		snprintf(aip_info->power_limit, HL_FIELD_MAX_SIZE, "%d", power);
+		snprintf(aip_info->eeprom_power_max, HL_FIELD_MAX_SIZE, "%d", power);
 	}
 	layout_field = find_field(aip_info->layout, "Max Frequency [MHz]");
 	if (layout_field)
diff --git a/src/hlml.c b/src/hlml.c
index 2c0608a..661de2f 100644
--- a/src/hlml.c
+++ b/src/hlml.c
@@ -787,16 +787,16 @@
 					unsigned int *default_limit)
 {
 	struct hl_smi_aip_info *aip_info = (struct hl_smi_aip_info *)device;
-	unsigned int power_limit_w;
+	unsigned int power_max_w;
 
 	INIT_CHECK
 
 	if (!aip_info || !default_limit)
 		return HLML_ERROR_INVALID_ARGUMENT;
 
-	HLML_LOG_DBG("Power limit is %s Watt", aip_info->power_limit);
-	power_limit_w = strtol(aip_info->power_limit, NULL, 10);
-	*default_limit = WATT_TO_MW(power_limit_w);
+	HLML_LOG_DBG("Power default limit is %s Watt", aip_info->eeprom_power_max);
+	power_max_w = strtol(aip_info->eeprom_power_max, NULL, 10);
+	*default_limit = WATT_TO_MW(power_max_w);
 
 	return HLML_SUCCESS;
 }
@@ -1970,7 +1970,7 @@
 	if (aip == NULL)
 		return HLML_ERROR_INVALID_ARGUMENT;
 
-	power_limit_mw_from_device = (unsigned int)WATT_TO_MW(strtoul(aip->power_limit, NULL, 10));
+	power_limit_mw_from_device = (unsigned int)WATT_TO_MW(strtoul(aip->eeprom_power_max, NULL, 10));
 	if (power_limit_mw_from_device && power_limit_mw_from_device < limit) {
 		HLML_LOG_ERR("can't set limit power to %uw, the max limit power is %uw\n", limit, power_limit_mw_from_device);
 		return HLML_ERROR_INVALID_ARGUMENT;
